torch/_inductor/pattern_matcher.py

```diff
@@ -236,9 +236,13 @@ def replace_by_example(
                 replacement graph.
 
         """
-        from torch._inductor.virtualized import V
+        from torch._inductor.virtualized import NullHandler, V
 
-        context = V.fake_mode if V.fake_mode is not None else contextlib.nullcontext
+        context = (
+            V.fake_mode
+            if not isinstance(V.fake_mode, NullHandler)
+            else contextlib.nullcontext()
+        )
```

Comment:
What happened here?

Reply:
V.fake_mode when its none it return NullHandler not none. I can make it or
but am unsure if there a situation were V.fake_mode is None

Reply:
this will be splitted to the other diff